src/daft-writers/src/lib.rs

```diff
@@ -61,26 +60,22 @@ pub trait WriterFactory: Send + Sync {
 
 pub fn make_physical_writer_factory(
     file_info: &OutputFileInfo,
-    schema: &SchemaRef,
     cfg: &DaftExecutionConfig,
 ) -> Arc<dyn WriterFactory<Input = Arc<MicroPartition>, Result = Vec<Table>>> {
-    let estimated_row_size_bytes = schema.estimate_row_size_bytes();
     let base_writer_factory = PhysicalWriterFactory::new(file_info.clone());
     match file_info.file_format {
         FileFormat::Parquet => {
-            let (target_file_rows, target_row_group_rows) = calculate_target_parquet_rows(
-                estimated_row_size_bytes,
-                cfg.parquet_target_filesize as f64,
-                cfg.parquet_target_row_group_size as f64,
-                cfg.parquet_inflation_factor,
-            );
+            let (target_in_memory_file_size, target_in_memory_row_group_size) =
```

Comment:
discussed offline but we can expose a `tell` method on the writer trait and update these values as we write.

Reply:
Nice suggestion, implemented. Works for s3 as well